tests/ASN1/ObjectTest.php

```diff
@@ -230,4 +230,98 @@ public function testFromBinaryThrowsException()
         $offset = 10;
         Object::fromBinary($binaryData, $offset);
     }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 0: Can not parse binary from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithEmptyStringThrowsException()
+    {
+        $data = '';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 2: Can not parse binary from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithSpacyStringThrowsException()
+    {
+        $data = '  ';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 1: Can not parse content length from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithNumberStringThrowsException()
+    {
+        $data = '1';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 25: Can not parse content length from data: Offset index larger than input size
+     * @depends testFromBinary
+     */
+    public function testFromBinaryWithGarbageStringThrowsException()
+    {
+        $data = 'certainly no asn.1 object';
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 2: Context-Specific explicitly tagged object [1] starting at offset 2 is shorter than required in the outer tag
+     * @depends testFromBinary
+     */
+    public function testFromBinaryExTagObjWithInvalidOuterLengthThrowsException1()
+    {
+        $data = hex2bin('a104040101');
+        //                  ^- this is wrong. correct would be "3"
+        Object::fromBinary($data);
+    }
+
+    /**
+     * @expectedException \FG\ASN1\Exception\ParserException
+     * @expectedExceptionMessage ASN.1 Parser Exception at offset 2: Context-Specific explicitly tagged object [1] starting at offset 2 is shorter than required in the outer tag
+     * @depends testFromBinary
+     */
+    public function testFromBinaryExTagObjWithInvalidOuterLengthThrowsException2()
```

Comment:
Its nice you cover all exceptions with a test. I guess you actually wanted to test the `longer than allowed` part in one of the `testFromBinaryExTagObjWithInvalidOuterLengthThrowsException` tests?

Otherwise I dont see the difference between them